tensorflow/core/grappler/op_types.cc

```diff
@@ -53,8 +53,8 @@ bool IsAnyDiv(const NodeDef& node) {
 
 bool IsAnyMatMul(const NodeDef& node) {
   const auto& op = node.op();
-  return op == "MatMul" || op == "BatchMatMul" || op == "SparseMatMul" ||
-         IsQuantizedMatMul(node);
+  return op == "MatMul" || op == "BatchMatMul" || op == "BatchMatMulV2" ||
```

Comment:
I think you might also have to change [the code here](https://github.com/tensorflow/tensorflow/blob/a12d05f82de28dceb62eaf4bb4b8b9daf3a3057b/tensorflow/core/grappler/optimizers/arithmetic_optimizer.cc#L2244)  to also handle BatchMatMulV2, and similar for the other two places in the function that use BatchMatMul. Also I'm not sure whether simply changing the BatchMatMul checks to also accept BatchMatMulV2 will work, since BatchMatMulV2 also supports broadcasting. /CC @rmlarsen, do you know if there will need to be special logic to handle broadcasting?

And BatchMatMul V2 should be added to [this test](https://github.com/tensorflow/tensorflow/blob/a12d05f82de28dceb62eaf4bb4b8b9daf3a3057b/tensorflow/core/grappler/optimizers/arithmetic_optimizer_test.cc#L821).

Reply:
:+1: Sorry, I missed that
I adapted the usage and added `BatchMatMulV2` to the test in 5cced167d795d71b7145a415479d89a2156dda76

Reply:
Thanks! Looking at the code more carefully, your logic will support broadcasting with BatchMatMulV2.